### PR TITLE
[Mobile App] Share feature enhancement

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- Enhancements to the share feature.

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

#### Screenshots (optional)

IMG_FFF153BF22A4-1

### CODE REVIEW

**Review by Mnoble-19 — APPROVED** [review]
Tested on Android LGTM!

**Comment by BenjaminSsempala** [comment]
@NoahNsimbe I noticed the share button copies the link to the clipboard but does not redirect to any app, Is it possible to make it in a way that it copies the link and then prompts you to choose the app to share to?(Like how the youtube sharing thing works).
I'm thinking that might encourage the user to actually share, and reduce instances where the user copies the link but forgets to share

**Comment by NoahNsimbe** [comment]
> @NoahNsimbe I noticed the share button copies the link to the clipboard but does not redirect to any app, Is it possible to make it in a way that it copies the link and then prompts you to choose the app to share to?(Like how the youtube sharing thing works). I'm thinking that might encourage the user to actually share, and reduce instances where the user copies the link but forgets to share

Thanks @BenjaminSsempala  I have made the changes

**Review by BenjaminSsempala — APPROVED** [review]

**Review by Mnoble-19 — APPROVED** [review]

File: mobile/lib/services/native_api.dart [hunk 1]
```diff
@@ -63,76 +64,174 @@ class RateService {
...
+      shareImage = Uri.parse(kya.imageUrl);
+    }
+
+    const uriPrefix = 'https://airqo.page.link';
+
+    PackageInfo packageInfo = await PackageInfo.fromPlatform();
+
+    final DynamicLinkParameters dynamicLinkParams = DynamicLinkParameters(
+      link: Uri.parse('https://airqo.net/?$params'),
+      uriPrefix: uriPrefix,
```
Comment (NoahNsimbe) [inline mobile/lib/services/native_api.dart:120]: creating share links

File: mobile/lib/screens/kya/kya_lessons_page.dart [hunk 2]
```diff
@@ -54,33 +53,70 @@ class _KyaLessonsPageState extends State<KyaLessonsPage> {
...
+                }
+                if (snapshot.hasData) {
+                  return InkWell(
+                    onTap: () async {
+                      Uri? link = snapshot.data;
+                      if (link != null) {
+                        await ShareService.shareLink(
+                          link,
+                          kya: widget.kya,
+                        );
```
Comment (NoahNsimbe) [inline mobile/lib/screens/kya/kya_lessons_page.dart:85]: sharing KYA using a link

File: mobile/lib/screens/insights/insights_widgets.dart [hunk 3]
```diff
@@ -1387,50 +1384,112 @@ class InsightsActionBar extends StatefulWidget {
...
+                if (snapshot.hasData) {
+                  Uri? link = snapshot.data;
+                  if (link != null) {
+                    return OutlinedButton(
+                      style: _leftButtonStyle,
+                      onPressed: () async {
+                        await ShareService.shareLink(
+                          link,
+                          airQualityReading: widget.airQualityReading,
+                        );
```
Comment (NoahNsimbe) [inline mobile/lib/screens/insights/insights_widgets.dart:1446]: sharing air quality using a link

**Pull request merged**